google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/CreateCampaignExperiment.java

```diff
@@ -0,0 +1,168 @@
+import com.google.ads.googleads.v3.utils.ResourceNames;
+import com.google.api.gax.longrunning.OperationFuture;
+import com.google.protobuf.Empty;
+import com.google.protobuf.Int64Value;
+import com.google.protobuf.StringValue;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+
+/** Adds a campaign experiment for a draft campaign. */
+public class CreateCampaignExperiment {
+  public static class CreateCampaignExperimentParams extends CodeSampleParams {
+
+    @Parameter(names = ArgumentNames.CUSTOMER_ID, required = true)
+    public long customerId;
+
+    @Parameter(names = ArgumentNames.CAMPAIGN_ID, required = true)
+    public long baseCampaignId;
+
+    @Parameter(names = ArgumentNames.DRAFT_ID, required = true)
+    public long draftId;
+  }
+
+  public static void main(String[] args) {
+    CreateCampaignExperimentParams params = new CreateCampaignExperimentParams();
+    if (!params.parseArguments(args)) {
+
+      // Either pass the required parameters for this example on the command line, or insert them
+      // into the code here. See the parameter class definition above for descriptions.
+      params.customerId = Long.parseLong("INSERT_CUSTOMER_ID_HERE");
+      params.baseCampaignId = Long.valueOf("INSERT_CAMPAIGN_ID_HERE");
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new CreateCampaignExperiment()
+          .runExample(googleAdsClient, params.customerId, params.baseCampaignId, params.draftId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /** Runs the example. */
+  private void runExample(
+      GoogleAdsClient googleAdsClient, long customerId, long baseCampaignId, long draftId) {
+    // Creates the campaign experiment.
+    CreateCampaignExperimentMetadata metadata =
+        createExperiment(googleAdsClient, customerId, baseCampaignId, draftId);
```

Comment:
The slight misalignment of comment, method name, and return type could be confusing. Is there a way to capture both? createExperimentAndReturnMetadata seems unwieldy. At a minimum, let's update the comment here and in the javadoc of the method.

Reply:
I think it's clear enough that we're returning metadata for the experiment... could you clarify what changes you would want to the comment/javadoc please?

Reply:
Right.